diag_manager/fms_diag_file_object.F90

```diff
@@ -574,8 +578,9 @@ subroutine add_start_time(this, start_time)
     !! simply update it with the start_time and set up the *_output variables
     this%start_time = start_time
     this%last_output = start_time
-    this%next_output = diag_time_inc(start_time, this%get_file_freq(), this%get_file_frequnit())
-    this%next_next_output = diag_time_inc(this%next_output, this%get_file_freq(), this%get_file_frequnit())
+    !TODO circular dependency
```

Comment:
What is this circular dependency?

Reply:
It is the same circular dependency as above. `diag_time_inc` is in `diag_utils_mod`

`diag_utils_mod` depends on `diag_axis_mod`

`diag_axis_mod` depends on `fms_diag_object_mod`

and `fms_diag_object_mod` depends on `fms_diag_file_object_mod`

It's a circle. 

We will need to move `diag_time_inc` to a lower level module that only depends on diag_data_mod.